### PR TITLE
Expose the `Error` type to allow users to set custom error handlers

### PR DESCRIPTION
**Pull request by LukeMathWalker**
Expose the `Error` type in the `global` module to allow users to define their own error handlers.  
This is currently impossible because `Error` is private and therefore users cannot provide a function that satisfies the constraints of `set_error_handler`.

### CODE REVIEW

File: opentelemetry/src/global/error_handler.rs [hunk 1]
```diff
@@ -18,12 +18,15 @@ pub enum Error {
     #[cfg(feature = "trace")]
     #[cfg_attr(docsrs, doc(cfg(feature = "trace")))]
     #[error(transparent)]
+    /// Failed to export traces.
     Trace(#[from] TraceError),
     #[cfg(feature = "metrics")]
     #[cfg_attr(docsrs, doc(cfg(feature = "metrics")))]
     #[error(transparent)]
+    /// Failed to export metrics.
```
Comment (TommyCpp) [inline opentelemetry/src/global/error_handler.rs:26]: I think [`MetricsError`](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry/src/metrics/mod.rs#L44) covers more than export errors. It also covers some configuration errors so maybe rephrase it into something like `errors from metrics module`  would be more accurate.

**Review by TommyCpp — APPROVED** [review]
Good catch. Thanks! Just a small suggestion on the comment

File: opentelemetry/src/global/error_handler.rs (hunk 1 above)
Comment (LukeMathWalker) [reply opentelemetry/src/global/error_handler.rs]: Updated 👍🏼
